src/client/SignIn.ts

```diff
@@ -22,6 +23,15 @@ export async function SignIn(username?: string, password?: string, useAuthToken?
         return Promise.resolve(new Big(401));
     }
 
+    if (typeof useAuthToken === "undefined") {
+        return new Promise(resolve => {
+            const onSuccess = (): void => resolve(new Big(200));
+            const onError = (error: { status: number }): void => resolve(new Big(error.status));
```

Comment:
Could you avoid duplication by extracting these onSuccess and onError handlers?

Reply:
Fixed